### ISSUE TITLE
Sources don't update if "winget source update" is run from an Admin CMD window in a "Standard" user account.

### ISSUE THREAD

**Issue by DrewNaylor** [microsoft/winget-cli#528]
# Brief description of your issue
Updating winget's sources from an Admin command prompt/PowerShell window while logged into a "Standard"/regular user account doesn't actually update the sources.

# Steps to reproduce

**Note:** winget will need to be installed in both accounts. Installing it in the Admin account will ask you to re-install, but in reality it's not installed in that account yet. winget may also complain about sources being broken after re-installing and trying to run from the "Standard" account, but I haven't noticed this while verifying this issue.

**Note 2:** There will need to be a "Standard" account in addition to an account with Administrator permissions, which is being referred to here as "Admin" to make it shorter.

1. Open CMD/PowerShell as Admin in a "Standard" account.
2. Run `winget source update`
3. Show or install a package that's been recently updated but hasn't had the new version cached in your local copy of the sources list. I personally ran into this issue with guinget, where winget said the latest version was 0.1.0.1 even though 0.1.1 was available through winget.
4. Open CMD/PowerShell without Admin permissions.
5. Run `winget source update`
6. Show or install the same package from step 3 and compare the versions displayed.

# Expected behavior
The sources should be updated just like they would be if you used a non-Admin command prompt/PowerShell window.

# Actual behavior
Sources don't actually update even though winget says they were updated. If you update the sources in a non-Admin window then show/install a package in the Admin window, it should display the new version based on my testing.

# Environment

```none
[winget --info]
Windows Package Manager v0.1.41821 Preview
Windows: Windows.Desktop v10.0.18363.959
Package: Microsoft.DesktopAppInstaller v1.0.41821.0

Any other software? None relevant to this issue.
```
I'd verify this in winget v0.1.42101 but my regular install has winget version v0.1.41331 installed in my regular account with v0.1.42101 in my Admin account on my main computer (running Windows 10 Version 10.0.18362.959) and I'm scared to update it as I'm having an issue in a VM described at the bottom of #502 that'll be posted soon.

**Comment by megamorf** [comment]
You don't actually need to update sources manually. That's being done behind the scenes after the cache has gone stale.
https://github.com/microsoft/winget-cli/blob/master/doc/Settings.md#autoupdateintervalinminutes

**Comment by DrewNaylor** [comment]
I'm aware of that, but unless it updates for the Admin account at the same time as the standard account, running `winget show (package)` as admin didn't update the cache for me even though the cache-updating progress bar showed up.

**Comment by megamorf** [comment]
As for the account issue. If you're running winget with a user that is not member of the Administrators group then that behaviour is expected. 

When a user account that is member of the Administrators group logs on, the Explorer and other processes will be started with a standard user access token. That user can then run a process elevated via UAC which grants that process an additional full administrative access token. Both belong to the same user and therefore same user profile.

What you have is a user that is not a member of the Administrators group. That means that user cannot elevate processes. In this case UAC allows you to use another user account that is member of the Administrators group to run that process. In this case you're actually executing a process in the context of a different user and therefore user profile. That means the paths where that foreign process writes to is different from your non-admin user and there is no way of solving this. That's by design

**Comment by DrewNaylor** [comment]
That makes sense, but the issue is that winget isn't updating the cache when I'm running it elevated from a regular account, while it does update when not running it elevated from the same regular account.

**Comment by megamorf** [comment]
You really need to be more specific with your statements and re-read my comment that explains how UAC works.

**Comment by DrewNaylor** [comment]
What I'm trying to say is, the elevated prompt in a standard account isn't updating the cache in the account the elevated prompt is using, while the non-elevated prompt is updating the cache in both the elevated and non-elevated prompts. I re-read the earlier comment more thoroughly and understand that part.

**Comment by megamorf** [comment]
Stop using the term "standard account" or "regular account". There is no such thing. Your user account is either member of the administrators group (default) or it isn't (either deliberate choice or corporate policy). The former can elevate its own processes whereas the latter needs another user account to do that.

**Comment by DrewNaylor** [comment]
That is correct. Sorry, I was just going off what Windows 10 calls accounts in the account settings section of the Settings app.

Decided to send `winget show`'s out put to a text file from both the elevated and non-elevated prompts, and it includes the progress bar when it updates its cache, so I'll edit the first post with them if it helps this issue.

**Comment by megamorf** [comment]
After clarifying your setup the problem is not with elevated permissions (CMD and its child processes having the administrative user token) but that you're simply using winget with two different user accounts.

That behaviour is expected. Here's a more detailed explanation why:

***

Say we have two Windows user accounts:
* **LimitedUser** (not member of `Administrators`)
* **NormalUser**

When I print `LimitedUser` path variables in PowerShell it gives:

```
PS> $env:UserProfile
C:\Users\LimitedUser\

PS> $env:LocalAppData
C:\Users\LimitedUser\AppData\Local
```

Now as `LimitedUser` I want to run PowerShell elevated. A UAC prompt shows up allowing me to run the process as `NormalUser`. When I run the same the output is different:

```
PS> $env:UserProfile
C:\Users\NormalUser\

PS> $env:LocalAppData
C:\Users\NormalUser\AppData\Local
``` 

Winget writes to `%LocalAppData%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe`. That means that depending on whether you ran winget in the first or second example its filesystem interaction was with different user profiles.

There's nothing to fix here - this is by design. Regardless of Windows version the user profile is considered an access boundary. This applies to all processes, e.g. an MSI setup will always write to %TEMP% and applied to your scenario you wouldn't find its files in `C:\Users\LimitedUser\AppData\Local\Temp` but under `C:\Users\NormalUser\AppData\Local\Temp`.

**Comment by DrewNaylor** [comment]
This makes sense regarding file access. Just strange that the progress bar behaves differently when run elevated from a user not in the Administrators group compared to running non-elevated from the same user.

**Comment by denelon** [comment]
@DrewNaylor, I will dig in a bit deeper to try and clarify exactly what is happening.

**Comment by denelon** [comment]
I believe @megamorf correctly described the account specific behavior. This looks like an edge case where there were multiple instances of installation for the client. The TTL value is stored based on the installed user(s). Each installed location has it's own reference timestamp for the TTL calculation. I expect the undesired behavior is being network inefficient primarily. We're looking at Delivery Optimization and being Metered Network aware in the backlog. I'm going to reference #151 as a signal to look back at this Issue. The fix for this would likely require some additional engineering to minimize the additional unnecessary downloads for the cache.
